go/vt/mysqlctl/builtinbackupengine.go

```diff
@@ -385,9 +391,11 @@ func (be *BuiltinBackupEngine) executeFullBackup(ctx context.Context, params Bac
 	}
 
 	// And set read-only mode
-	params.Logger.Infof("resetting mysqld read-only to %v", readOnly)
-	if err := params.Mysqld.SetReadOnly(readOnly); err != nil {
-		return usable, err
+	if resetSuperReadOnlyFunc != nil {
```

Comment:
I think we don't need this anymore but I am keeping it because I am not sure why on first place this logic of switching read_only was inserted. Primary should alway be non_read_only.

Reply:
This code block is not gated by `sourceIsPrimary`. The previous authors are just making sure this is set to whatever it was before the backup was started. It could have been done differently, basically something like
```
if sourceIsPrimary {
    ... params.Mysqld.SetReadOnly(false) ...
    ...
}
```

Reply:
I will just then create parity with existing code then.